*** README.md	2016-04-20 10:13:44.319573454 -0700
--- local.md	2016-04-27 13:44:58.045974793 -0700
***************
*** 7,12 ****
--- 7,21 ----
  
  ## Lab
  
+ ### Mac / Windows
+ 
+ You'll need to know the IP of the `docker-machine` vm that is your docker host:
+ 
+ ```
+ docker-machine ip $(docker-machine active)
+ ```
+ 
+ Use this when browsing to a node IP in place of `localhost` below.
  
  ### Pods
  
***************
*** 47,62 ****
  ### Service
  
  To access Lobsters from outside the cluster, we'll need a service. The
! service defined in [service.yaml](service.yaml) will route traffic to
! any pod with the label `app: lobsters`, which matches our pod
! definition. The service is for port 80, but routes to the port labeled
! `web` in our pod definition. The `type: LoadBalancer` creates an IP
! external to the cluster in supported environments.
  
  Create the service and pod:
  
  ```
! kubectl create -f ./service.yaml,./pod.yaml
  ```
  
  ```
--- 56,71 ----
  ### Service
  
  To access Lobsters from outside the cluster, we'll need a service. The
! service defined in [service-local.yaml](service-local.yaml) will route
! traffic to any pod with the label `app: lobsters`, which matches our
! pod definition. The service routs to the port labeled `web` in our pod
! definition. The `type: NodePort` line allows traffic on a particular
! port of each node to be routed to the service.
  
  Create the service and pod:
  
  ```
! kubectl create -f ./service-local.yaml,./pod.yaml
  ```
  
  ```
***************
*** 64,82 ****
  pod "lobsters" created
  ```
  
! Wait for the external IP:
  
  ```
! kubectl get svc lobsters
  ```
  
  ```
! NAME       CLUSTER-IP     EXTERNAL-IP    PORT(S)   AGE
! lobsters   10.3.253.158   1.2.3.4        80/TCP    1m
  ```
  
! Check that it is working by visiting the external IP in your browser.
! 
  
  
  Delete
--- 73,90 ----
  pod "lobsters" created
  ```
  
! Check the service's node port, yours will be different:
  
  ```
! kubectl get svc lobsters -o yaml | grep nodePort
  ```
  
  ```
!   - nodePort: 31618
  ```
  
! Check that it is working by visiting the node IP with the port you
! found `http://localhost:31618/`
  
  
  Delete
***************
*** 105,111 ****
  Start lobsters using an RC, use the same service definition:
  
  ```
! kubectl create -f ./rc.yaml,./service.yaml
  ```
  
  ```
--- 113,119 ----
  Start lobsters using an RC, use the same service definition:
  
  ```
! kubectl create -f ./rc.yaml,./service-local.yaml
  ```
  
  ```
***************
*** 113,130 ****
  service "lobsters" created
  ```
  
! Wait for the external IP:
  
  ```
! kubectl get svc lobsters
  ```
  
  ```
! NAME       CLUSTER-IP     EXTERNAL-IP    PORT(S)   AGE
! lobsters   10.3.253.158   1.2.3.4        80/TCP    1m
  ```
  
! Check that it is working by visiting the external IP in your browser.
  
  
  Now, look at the pod
--- 121,138 ----
  service "lobsters" created
  ```
  
! Check the service's node port, yours will be different:
  
  ```
! kubectl get svc lobsters -o yaml | grep nodePort
  ```
  
  ```
!   - nodePort: 31618
  ```
  
! Check that it is working by visiting the node IP with the port you
! found `http://localhost:31618/`
  
  
  Now, look at the pod
***************
*** 134,141 ****
  ```
  
  ```
! NAME             READY     STATUS    RESTARTS   AGE       NODE
! lobsters-jf0xs   1/1       Running   0          2m        gke-myclus-2f1fdf58-node-lfaa
  ```
  
  This pod was created by the replication controller. Try deleting the
--- 142,149 ----
  ```
  
  ```
! NAME                   READY     STATUS    RESTARTS   AGE       NODE
! lobsters-tx1sa         1/1       Running   0          21s       127.0.0.1
  ```
  
  This pod was created by the replication controller. Try deleting the
***************
*** 156,166 ****
  ```
  
  ```
! NAME             READY     STATUS    RESTARTS   AGE       NODE
! lobsters-t1vwk   1/1       Running   0          6s        gke-myclus-2f1fdf58-node-lfaa
  ```
  
! A new pod was created! It might even be on a different node.
  
  Scaling is as easy as:
  
--- 164,174 ----
  ```
  
  ```
! NAME                   READY     STATUS    RESTARTS   AGE       NODE
! lobsters-l5fq3         1/1       Running   0          1s        127.0.0.1
  ```
  
! A new pod was created!
  
  Scaling is as easy as:
  
***************
*** 179,190 ****
  ```
  
  ```
! NAME             READY     STATUS              RESTARTS   AGE       NODE
! lobsters-32ona   1/1       Running             0          26s       gke-myclus-2f1fdf58-node-lfaa
! lobsters-8twm0   1/1       Running             0          2m        gke-myclus-2f1fdf58-node-lfaa
! lobsters-hhves   0/1       ContainerCreating   0          26s       gke-myclus-2f1fdf58-node-kxe4
! lobsters-lv5km   0/1       ContainerCreating   0          26s       gke-myclus-2f1fdf58-node-bvxp
! lobsters-tlojp   0/1       ContainerCreating   0          26s       gke-myclus-2f1fdf58-node-bvxp
  ```
  
  Also the RC
--- 187,198 ----
  ```
  
  ```
! NAME                   READY     STATUS    RESTARTS   AGE       NODE
! lobsters-9ijsi         1/1       Running   0          6s        127.0.0.1
! lobsters-l5fq3         1/1       Running   0          36s       127.0.0.1
! lobsters-pfnlj         1/1       Running   0          6s        127.0.0.1
! lobsters-sceuy         1/1       Running   0          6s        127.0.0.1
! lobsters-txgwb         1/1       Running   0          6s        127.0.0.1
  ```
  
  Also the RC
***************
*** 230,236 ****
  RC declaration.
  
  ```
! kubectl create -f ./dep.yaml,./service.yaml
  ```
  
  ```
--- 238,244 ----
  RC declaration.
  
  ```
! kubectl create -f ./dep.yaml,./service-local.yaml
  ```
  
  ```
***************
*** 292,295 ****
  ```
  kubectl delete pod,rc,svc,deployment -l app=lobsters
  ```
- 
--- 300,302 ----
